vlo-web-app/src/main/java/eu/clarin/cmdi/vlo/wicket/provider/SolrDocumentExpansionPairProvider.java

```diff
@@ -22,36 +22,84 @@
 import eu.clarin.cmdi.vlo.pojo.QueryFacetsSelection;
 import eu.clarin.cmdi.vlo.service.solr.SolrDocumentExpansionList;
 import eu.clarin.cmdi.vlo.service.solr.SolrDocumentExpansionPair;
+import eu.clarin.cmdi.vlo.service.solr.impl.SolrDocumentExpansionPairImpl;
 import eu.clarin.cmdi.vlo.service.solr.SolrDocumentService;
 import eu.clarin.cmdi.vlo.wicket.model.SolrDocumentExpansionPairModel;
 import java.math.BigDecimal;
+import java.util.ArrayList;
 import java.util.Iterator;
+import java.util.List;
+
 import org.apache.wicket.markup.repeater.data.IDataProvider;
 import org.apache.wicket.model.IModel;
 
+import eu.clarin.cmdi.vlo.VloWebSession;
+import org.apache.wicket.protocol.http.request.WebClientInfo;
+import org.apache.wicket.request.cycle.RequestCycle;
+import org.slf4j.LoggerFactory;
+
+import com.sun.istack.logging.Logger;
+
+import org.apache.wicket.protocol.http.servlet.ServletWebRequest;
+import eu.clarin.cmdi.vlo.pojo.QueryFacetsSelection;
+import eu.clarin.cmdi.vlo.exposure.models.SearchQuery;
+import eu.clarin.cmdi.vlo.exposure.models.SearchResult;
+import eu.clarin.cmdi.vlo.config.ServletVloConfigFactory;
+import eu.clarin.cmdi.vlo.config.VloConfig;
+
 /**
  *
  * @author twagoo
  */
 public class SolrDocumentExpansionPairProvider implements IDataProvider<SolrDocumentExpansionPair> {
+    private final static org.slf4j.Logger logger = LoggerFactory.getLogger(SolrDocumentExpansionPairProvider.class);
 
     private final FieldNameService fieldNameService;
 
     private final IModel<QueryFacetsSelection> selectionModel;
 
     private Long size;
 
-    public SolrDocumentExpansionPairProvider(IModel<QueryFacetsSelection> selection, FieldNameService fieldNameService) {
+    public SolrDocumentExpansionPairProvider(IModel<QueryFacetsSelection> selection,
+            FieldNameService fieldNameService) {
         this.selectionModel = selection;
         this.fieldNameService = fieldNameService;
     }
 
     @Override
     public Iterator<? extends SolrDocumentExpansionPair> iterator(long first, long count) {
-        final SolrDocumentExpansionList documents = getDocumentService().getDocumentsWithExpansion(selectionModel.getObject(),
-                BigDecimal.valueOf(first).intValueExact(), // safe long->int conversion
-                BigDecimal.valueOf(count).intValueExact(),
-                FacetConstants.COLLAPSE_FIELD_NAME); // safe long->int conversion
+        final SolrDocumentExpansionList documents = getDocumentService().getDocumentsWithExpansion(
```

Comment:
I think this is a fairly large amount of logic that is not at the core of what this class (expansion pair provider) is primarily for. Would it possible to encapsule the logic in either a method of its own or perhaps even better an 'exposure tracker' service to make it easier to understand the different 'aspects'.